### PR TITLE
Ref #4220: Add the camel-swift extension

### LINKED ISSUE #4220: Camel 3.20 - camel-swift extension
There is a new component (dataformat)
https://github.com/apache/camel/tree/main/components/camel-swift

### PR DESCRIPTION
**Pull request by essobedo**
fixes #4220 

## Motivation

There is a new component (dataformat) called `camel-swift` that we would like to add to the project.

## Modifications:

* Creates the new extension thanks to the maven plugin
* Adds support of the JVM mode for both types of messages (MX and MT)
* Adds support of the native mode to MT messages only because MX messages bring too many classes, methods, and fields to access by reflection which slows down the native image build

### CODE REVIEW

**Comment by essobedo** [comment]
Proposed as a draft for now, because first, it requires Camel 3.20 and it raises a potential bug in Quarkus.

**Comment by essobedo** [comment]
The related bug in Quarkus https://github.com/quarkusio/quarkus/issues/29693

**Comment by essobedo** [comment]
The bug in Quarkus is finally fixed but in Quarkus 3 which requires the Jakarta migration and unfortunately, the swift library has not been migrated yet so this PR is blocked

**Comment by ppalaga** [comment]
> The bug in Quarkus is finally fixed but in Quarkus 3 which requires the Jakarta migration and unfortunately, the swift library has not been migrated yet so this PR is blocked

Do you, by any chance, have an issue or PR ref. that is blocking this PR?

**Comment by essobedo** [comment]
Blocked by https://github.com/prowide/prowide-iso20022/issues/56

**Comment by zubri** [comment]
> Blocked by [prowide/prowide-iso20022#56](https://github.com/prowide/prowide-iso20022/issues/56)

We are working on it at Prowide, the beta release should be published soon, within one or two weeks.

**Comment by essobedo** [comment]
@zubri great news thanks for sharing

**Comment by essobedo** [comment]
NB: The examples are failing but it is not related to those changes since [it also fails on the camel-main branch](https://github.com/apache/camel-quarkus/actions/runs/5186820109/jobs/9349016573).

**Comment by essobedo** [comment]
Any remarks/suggestions before I merge it?

File: integration-tests/swift/pom.xml
Comment (essobedo) [reply integration-tests/swift/pom.xml:71]: This is needed by prowide libs to serialize MX messages in JSON. I guess it is not expected so I added a comment about it https://github.com/prowide/prowide-iso20022/pull/83#issuecomment-1580150900

Comment (essobedo) [reply integration-tests/swift/pom.xml:71]: Let's see what will be their answer but hopefully, it is a temporary workaround

**Comment by essobedo** [comment]
@jamesnetherton thx for your feedback, I think I addressed all your remarks

File: integration-tests/swift/src/test/java/org/apache/camel/quarkus/component/swift/mt/it/SwiftMtTest.java
Comment (jamesnetherton) [reply integration-tests/swift/src/test/java/org/apache/camel/quarkus/component/swift/mt/it/SwiftMtTest.java]: I don't think reading from `Paths.get("target/test-classes/*")` will work either.

When the test runs in the Quarkus Platform, it's via the `test-jar` artifact. So there will be a `target` directory, but not `target/test-classes`.

Hence, it's safer to read from the classpath directly or copy the content from the classpath into `target` if you really must read from the filesystem. Hope that makes sense.

Comment (essobedo) [reply integration-tests/swift/src/test/java/org/apache/camel/quarkus/component/swift/mt/it/SwiftMtTest.java]: I better understand, thx for the clarifications. It should be ok now

**Review by jamesnetherton — APPROVED** [review]

File: integration-tests/swift/pom.xml
Comment (ppalaga) [inline integration-tests/swift/pom.xml:126]:
```suggestion
                </dependency>
                <dependency>
                    <groupId>org.apache.camel.quarkus</groupId>
                    <artifactId>camel-quarkus-direct-deployment</artifactId>
                    <version>${project.version}</version>
                    <type>pom</type>
                    <scope>test</scope>
                    <exclusions>
                        <exclusion>
                            <groupId>*</groupId>
                            <artifactId>*</artifactId>
                        </exclusion>
                    </exclusions>
                </dependency>
```

**Review by ppalaga — APPROVED** [review]
Nice, thanks a lot, @essobedo!

File: integration-tests/swift/pom.xml
Comment (essobedo) [reply integration-tests/swift/pom.xml:139]: Thx @ppalaga, good catch, it should be good now

File: integration-tests/swift/pom.xml
Comment (essobedo) [reply integration-tests/swift/pom.xml:71]: The follow up PR https://github.com/apache/camel-quarkus/pull/4992

**Pull request merged**
